sos/report/plugins/ceph_mgr.py

```diff
@@ -103,17 +72,67 @@ def setup(self):
             "version"
         ]
 
+        directory = ''
+        if not microceph_pkg:
+            directory = '/var/run/ceph'
+            self.add_file_tags({
+                '/var/log/ceph/(.*/)?ceph-mgr.*.log': 'ceph_mgr_log',
+            })
+
+            self.add_forbidden_path([
+                "/etc/ceph/*keyring*",
+                "/var/lib/ceph/**/*keyring*",
+                "/var/lib/ceph/**/osd*",
+                "/var/lib/ceph/**/mon*",
+                # Excludes temporary ceph-osd mount location like
+                # /var/lib/ceph/tmp/mnt.XXXX from sos collection.
+                "/var/lib/ceph/**/tmp/*mnt*",
+                "/etc/ceph/*bindpass*",
+            ])
+
+            self.add_copy_spec([
+                "/var/log/ceph/**/ceph-mgr*.log",
+                "/var/lib/ceph/**/mgr*",
+                "/var/lib/ceph/**/bootstrap-mgr/",
+                "/run/ceph/**/ceph-mgr*",
+            ])
+
+        else:
+            directory = '/var/snap/microceph'
+            self.add_file_tags({
+                '/var/snap/microceph/common/logs/ceph-mgr.*.log':
+                'ceph_mgr_log',
+            })
+
+            self.add_forbidden_path([
+                "/var/snap/microceph/common/**/*keyring*",
+            ])
+
+            self.add_copy_spec([
+                "/var/snap/microceph/common/logs/ceph-mgr*.log",
+            ])
+
+        self.add_cmd_output(
+            [f"ceph {cmd}" for cmd in ceph_mgr_cmds])
+
+        # get ceph_cmds again as json for easier automation parsing
+        self.add_cmd_output(
+            [f"ceph {cmd} --format json-pretty" for cmd in ceph_mgr_cmds],
+            subdir="json_output",
+        )
+
         self.add_cmd_output([
-            f"ceph daemon {m} {cmd}" for m in self.get_socks() for cmd in cmds]
+            f"ceph daemon {m} {cmd}" for m in self.get_socks(
+                directory) for cmd in cmds]
         )
 
-    def get_socks(self):
+    def get_socks(self, directory):
         """
         Find any available admin sockets under /var/run/ceph (or subdirs for
         later versions of Ceph) which can be used for ceph daemon commands
         """
         ceph_sockets = []
-        for rdir, dirs, files in os.walk('/var/run/ceph/'):
+        for rdir, dirs, files in os.walk(directory):
```

Comment:
Is there a use case where `sos` could be run from within a container for a microceph environment? If so, we'd want to look at a path wrapper for Plugin (like we do for `Plugin.path_exists()`) to automatically account for any container sysroot.

If there isn't such a use case, no problem here.

Reply:
I don't think this is a valid, or at least a typical use case for microceph. Maybe @UtkarshBhatthere can confirm it.

There is however a ceph in containers product too at canonical (https://github.com/canonical/ceph-containers) which is either cephadm or rook based deployments. 

For cephadm based deployments sos works fine when run on the host because they're podman containers. For rook based deployments we still need to understand how to get sosreport working. (pasting a canonical only link for reference for canonical folks - https://warthogs.atlassian.net/browse/SET-70 )

Reply:
Hey Jake, MicroCeph running targets are (mainly) physical/virtual machines and system containers like LXC. This is not to say that one can't bring up a working microceph cluster using application containers like Docker or Podman, however, the benefits of such an approach are limited.

Reply:
based on Utkarsh's comment it seems OK to not consider that use-case for now.